lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1241,6 +1254,26 @@ private function class_name_updates_to_lexical_updates() {
 		}
 	}
 
+	/**
+	 * Converts attribute updates into lexical updates.
+	 *
+	 * This method is only meant to run right before the attribute updates are applied.
+	 * The behavior in all other cases is undefined.
+	 *
+	 * @return void
+	 * @since 6.2.0
+	 *
+	 * @see $attribute_updates
+	 * @see $lexical_updates
+	 */
+	private function attribute_updates_to_lexical_updates() {
+		$this->lexical_updates   = array_merge(
+			$this->lexical_updates,
+			array_values( $this->attribute_updates )
```

Comment:
on a side note, I think we're going to want to avoid all array functions as a general rule because of how poorly they perform compared to lower-level operations and iteration. as always, we can somewhat naively measure each approach and get a general feel that should be reasonable enough to lead our decisions.

Reply:
Yeah, I'm a bit disappointed at PHP's optimizing skills (or lack thereof) at this point 😬

Reply:
be5ab5f